File: notes/multistore.go
package notes

import (
	"github.com/karabatov/ddpub/config"
	"github.com/karabatov/ddpub/dd"
)

type MultiStore struct {
	Main      *Store
	SubStores map[dd.Language]*Store
}

func NewMultiStore(w *config.Website, notesDir string) (*MultiStore, error) {
	var m MultiStore

	mainStore, err := newStore(w.Main, notesDir)
	if err != nil {
		return nil, err
	}
	m.Main = mainStore

	m.SubStores = make(map[dd.Language]*Store)
	for _, cfg := range w.SubConfigs {
		s, err := newStore(cfg, notesDir)
		if err != nil {
			return nil, err
		}
		m.SubStores[cfg.Language.Code] = s
	}

	return &m, nil
}

File: src/notes/multistore.rs
//! MultiStore.

use crate::config::{Website, WebsiteLang};
use crate::dd::Language;
use crate::error::Result;
use crate::notes::Store;
use std::collections::HashMap;

pub struct MultiStore {
    pub main: Store,
    pub sub_stores: HashMap<Language, Store>,
}

impl MultiStore {
    pub fn new(w: &Website, notes_dir: &str) -> Result<Self> {
        let all_configs: Vec<&WebsiteLang> = std::iter::once(&w.main)
            .chain(w.sub_configs.iter())
            .collect();

        let main = Store::new(&w.main, notes_dir, &all_configs)?;

        let mut sub_stores = HashMap::new();
        for cfg in &w.sub_configs {
            let s = Store::new(cfg, notes_dir, &all_configs)?;
            sub_stores.insert(cfg.language.code, s);
        }

        Ok(MultiStore { main, sub_stores })
    }
}
